lib/validator.rb

```diff
@@ -0,0 +1,22 @@
+class SelectionValidator
```

Comment:
Nice! 

Here are a couple of questions I have about this class.

1. Does this class have multiple responsibilities?
   * It seems to me that this object is _validating_ and _parsing_. 
2. What is the public interface of this object? Is it cohesive?
   * How well do `valid`, `available`, and `input_to_integer` fit together in the domain of something that is _validating_?

Reply:
I kept the parsing function in human player, and just have one validate method for the validator class